device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/convention/DeviceCommand.java

```diff
@@ -0,0 +1,182 @@
+    PayloadConvention payloadConvention;
+
+    private final class deviceCommandResponseCallback implements MessageCallback
+    {
+        final DeviceClientConfig nestedConfig = config;
+
+        @Setter(AccessLevel.PRIVATE)
+        @Getter(AccessLevel.PRIVATE)
+        private PayloadConvention payloadConvention;
+
+        public deviceCommandResponseCallback(PayloadConvention payloadConvention)
+        {
+            setPayloadConvention(payloadConvention);
+        }
+
+        @Override
+        public IotHubMessageResult execute(Message message, Object callbackContext)
+        {
+            synchronized (DEVICE_METHOD_LOCK)
+            {
+                IotHubStatusCode iotHubStatus = IotHubStatusCode.ERROR;
+                IotHubMessageResult result = IotHubMessageResult.ABANDON;
+
+                if (message.getMessageType() != MessageType.DEVICE_METHODS)
+                {
+                    log.error("Unexpected message type received {}", message.getMessageType());
+                    deviceMethodStatusCallback.execute(iotHubStatus, deviceMethodStatusCallbackContext);
+                    return IotHubMessageResult.ABANDON;
+                }
+
+                IotHubTransportMessage methodMessage = (IotHubTransportMessage) message;
+
+                if (methodMessage.getDeviceOperationType() == DeviceOperations.DEVICE_OPERATION_METHOD_RECEIVE_REQUEST)
+                {
+                    if (deviceCommandCallback != null)
+                    {
+                        if (!isSubscribed)
+                        {
+                            isSubscribed = true;
+                        }
+                        try
+                        {
+                            String componentName = null;
+                            String commandName = methodMessage.getMethodName();
+
+                            // The current format for command names with components is componentName*commandName
+                            // This set of operations will attempt to split on the * character. A note for the future.
+                            // There is a consideration for multiple nested components so we will need to alter this section to support that.
+                            if (commandName.contains("*"))
+                            {
+                                String[] componentAndMethod = methodMessage.getMethodName().split("\\*");
+                                componentName = componentAndMethod[0];
+                                commandName = componentAndMethod[1];
+
+                            }
+
+                            log.trace("Executing command invocation callback for component {} with command name {} for message {}",  componentName == null ? "default" : componentName , commandName, methodMessage);
+                            DeviceCommandRequest commandRequest = new DeviceCommandRequest(componentName, commandName, methodMessage.getBytes(), payloadConvention);
```

Comment:
The device command request allows us to be specific with the payload. Therefore a customer can use the convention without being aware of it.